features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx

```diff
@@ -0,0 +1,184 @@
+                        <FormattedMessage
+                            id='Apis.Details.Documents.MarkdownEditor.edit.content.of'
+                            defaultMessage='Edit Description of '
+                        />
+                        {api.name}
+                    </Typography>
+                    <Button
+                        className={classes.button}
+                        variant='contained'
+                        disabled={isUpdating}
+                        color='primary'
+                        onClick={updateDescription}
+                    >
+                        <FormattedMessage
+                            id='Apis.Details.Documents.MarkdownEditor.update.content.button'
+                            defaultMessage='Update Description'
+                        />
+                        {isUpdating && <CircularProgress size={24} />}
+                    </Button>
+                    <Button className={classes.button} onClick={toggleOpen}>
+                        <FormattedMessage
+                            id='Apis.Details.Documents.MarkdownEditor.cancel.button'
+                            defaultMessage='Cancel'
+                        />
+                    </Button>
+                </Paper>
+                <div className={classes.splitWrapper}>
+                    <Grid container spacing={7}>
+                        <Grid item xs={6}>
+                            <Suspense fallback={<CircularProgress />}>
+                                <MonacoEditor
+                                    width='100%'
+                                    height='100vh'
+                                    language='markdown'
+                                    theme='vs-dark'
+                                    value={description}
+                                    options={{ selectOnLineNumbers: true }}
+                                    onChange={changeDescription}
+                                    editorDidMount={editorDidMount}
+                                />
+                            </Suspense>
+                        </Grid>
+                        <Grid item xs={6}>
+                            <div className={classes.markdownViewWrapper}>
+                                <Suspense fallback={<CircularProgress />}>
+                                    <ReactMarkdown escapeHtml={false} source={description} />
+                                </Suspense>
+                            </div>
+                        </Grid>
+                    </Grid>
+                </div>
+            </Dialog>
+        </div>
+    );
+}
+
+MarkdownEditor.propTypes = {
+    api: PropTypes.shape({}).isRequired,
+    configDispatcher: PropTypes.func.isRequired,
+    classes: PropTypes.shape({}).isRequired,
+    intl: PropTypes.shape({}).isRequired,
+};
+
+export default injectIntl(withRouter(withStyles(styles)(MarkdownEditor)));
```

Comment:
You might have taken this from an existing component, but this injects and higher-order components are a bit old with newer react hooks, Shall we replace `injectIntl` and `withRouter` , `withStyles` with their respective hooks

Reply:
there is no usage of `withRouter` HOC , so I think we can remove that as well

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9362/commits/f66e3a9a634e8447abebfb0698f063af3abb7f6b